src/components/Tasks/TaskListItem.tsx

```diff
@@ -36,26 +36,40 @@ interface PropsType {
   entry: TaskType;
   onClick: (task: TaskType) => void;
   onSave: (item: PimType, journalUid: string, originalItem?: PimType) => Promise<void>;
+  onUpdate: (message: string) => void;
```

Comment:
I really don't think this is the way to do it... First of all, `onUpdate` sounds like a generic name which can mislead you into thinking this is actually just a generic action, when in fact, the message parameter is very tied to what you are trying to do.
Also, the fact that you also only call this when rescheduling a task, and not really on an update, is quite misleading.
I don't think this is the right way to do it. You maybe just want a redux queue for toasts and a toast presenter somewhere high up, that just processeses the queue and presents toasts.

Reply:
I also felt that this was clunky but wasn't sure how to handle it. I like the idea of a redux queue for processing toasts. Maybe this is something that should happen in a separate PR then?

Reply:
Another idea I was thinking of is should the `TaskList` component just handle all the saving/updating of the items? That way any notification displaying can be contained within that component, and it also takes a lot of the functionality out of the `TaskListItems` and makes them purely presentational.

Reply:
Yeah, this idea sounds good too. I don't like the PIM abstraction I did anyway, it was stupid and was more trouble than it's worth. Feel free to change that.

Reply:
haha I don't mind the abstraction, but it has caused some issues. I don't think I'll go so far as reworking that in this PR though, but maybe it is something to think about for the future.